internal/action/actions.go

```diff
@@ -160,13 +160,19 @@ func (h *BufPane) Center() bool {
 	return true
 }
 
-// MoveCursorToView moves the cursor to the current view
-func (h *BufPane) MoveCursorToView() bool {
+// CursorToView moves the cursor to the current view
+func (h *BufPane) CursorToView() bool {
 	v := h.GetView()
-	h.Cursor.GotoLoc(buffer.Loc{0, v.StartLine.Line})
-	h.RemoveAllMultiCursors()
 	h.Cursor.Deselect(true)
-	h.Relocate()
+	loc := h.Cursor.Loc
+	if(loc.Y < v.StartLine.Line || loc.Y > v.StartLine.Line + h.BufView().Height - 1) {
+		scrollmargin := int(h.Buf.Settings["scrollmargin"].(float64))
+		h.Cursor.GotoLoc(h.LocFromVLoc(display.VLoc{
+			SLoc: h.Scroll(v.StartLine, scrollmargin),
+			VisualX: 0,
+		}))
```

Comment:
So this new `CursorToView` action means "move cursor to the current view" without defining where exactly will it move the cursor in the current view. And I'm not sure if the top of the view (plus scrollmargin) is the best "default" location.

In other cases when we move cursor to a location which is far from its current location, we adjust the view so that the cursor is at 25% of the view height. (See e.g. #2628 where it was first implemented.) So maybe for `CursorToView` it'd be also a good idea to put the cursor at 25% of the height of the current view, not near the top? What do you think?

Reply:
I think it might be good to make this a configurable option, because if the user opens a file and the cursor is at the first line, then if the user hits page down it moves to 25% from the top, it might be confusing. However, that might be a task for another PR. For now I'm moving it to 25%.

Reply:
> I think it might be good to make this a configurable option

Or maybe better a separate action `CursorToViewTop` (and maybe `CursorToViewBottom` as well), for better flexibility: the user could bind one key to `CursorToView` and another key to `CursorToViewTop`.

(Or if you'd like to be able to configure arbitrary line number or percentage within the current view, the most flexible way would be to add a new command, similar to `goto` and `jump` but for jumping to the given line within the current view. Since you can bind keys to command too, with arbitrary arguments.)

Reply:
BTW I've realized there was one more flaw in your previous implementation (and which we should consider if we implement `CursorToViewTop`): it always added scrollmargin to the top line, which is not what the user expects in the case when the view is at the very beginning of the file. In that case the user expects the cursor to be at the very top, right?

Reply:
That was a deliberate decision, not a flaw. The problem with moving the cursor to the very top is that if the user has `scrollmargin` set to anything over 0 and the cursor is moved to the top, the next time the user hits any cursor movement key, the view jumps to accommodate `scrollmargin`. This is jarring.